lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -3161,7 +3161,7 @@ namespace Js
                         // in EntryConcat like we do with Arrays because a getProperty on an object Length
                         // is observable. The result is we have to check for overflows separately for
                         // spreadable objects and promote to a bigger index type when we find them.
-                        ConcatArgs<BigIndex>(pDestArray, remoteTypeIds, args, scriptContext, idxArg, idxDest, /*firstPromotedItemIsSpreadable*/true, length);
+                        ConcatArgs<BigIndex>(pDestObj, remoteTypeIds, args, scriptContext, idxArg, idxDest, /*firstPromotedItemIsSpreadable*/true, length);
```

Comment:
>pDestObj [](start = 45, length = 8)

Should we check pDestArray and send pDestObj down the generic path down below instead?